cmd/parca-agent/main.go

```diff
@@ -106,6 +109,8 @@ type Profiler interface {
 	Name() string
 	Run(ctx context.Context) error
 
+	Labels(pid profiler.PID) model.LabelSet
+	Relabel(model.LabelSet) labels.Labels
```

Comment:
Actually, we don't need to expose the `Relabel` method at all. We can handle all of these inside the `Label` method. We should comment that we apply relabeling method inside this function and if it returns nil, it means the process needs to be dropped.

This would also prevent unnecessary conversion between label set and labels.

Reply:
This is done this way because `Labels()` and `Relabel()` are handle subtlety differently between the profiler and the UI to avoid unnecessary conversions, imo it is the most efficient as it is.

I plan to do some refactoring when I get to caching and reloading, hopefully it will be cleaner